lib/auth/auth.go

```diff
@@ -803,6 +885,34 @@ func (a *AuthServer) ValidateOIDCAuthCallback(q url.Values) (*OIDCAuthResponse,
 	return response, nil
 }
 
+func (a *AuthServer) DeleteRole(name string) error {
+	// check if this role is used by CA or Users
+	users, err := a.Identity.GetUsers()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, u := range users {
+		for _, r := range u.GetRoles() {
+			if r == name {
+				return trace.BadParameter("role %v is used by user %v", name, u.GetName())
+			}
+		}
+	}
+	// check if it's used by some cert authorities
+	cas, err := a.Trust.GetCertAuthorities(services.UserCA, false)
```

Comment:
This can happen when you give access to some other organisation to your cluster and this cert authority is related to the other org. Then you can assign local roles to users authenticating via remote certificate authority.